File: src/main/java/no/nav/fo/veilarbregistrering/oppgave/RoutingStep.java
package no.nav.fo.veilarbregistrering.oppgave;

import no.nav.fo.veilarbregistrering.metrics.Metric;

public enum  RoutingStep implements Metric {

    GeografiskTilknytning_Feilet,
    GeografiskTilknytning_Funnet,
    Enhetsnummer_Feilet,
    SisteArbeidsforhold_IkkeFunnet,
    OrgNummer_ikkeFunnet,
    OrgDetaljer_IkkeFunnet,
    Kommunenummer_IkkeFunnet,
    Enhetsnummer_IkkeFunnet,
    Enhetsnummer_Funnet;

    @Override
    public String fieldName() {
        return "routingStep";
    }

    @Override
    public Object value() {
        return this.toString();
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/oppgave/RoutingStep.kt
package no.nav.fo.veilarbregistrering.oppgave

import no.nav.fo.veilarbregistrering.metrics.Metric

@Suppress("EnumEntryName")
enum class RoutingStep : Metric {
    GeografiskTilknytning_Feilet,
    GeografiskTilknytning_Funnet,
    GeografiskTilknytning_ByMedBydel_Funnet,
    GeografiskTilknytning_Utland,
    Enhetsnummer_Feilet,
    SisteArbeidsforhold_IkkeFunnet,
    OrgNummer_ikkeFunnet,
    OrgDetaljer_IkkeFunnet,
    Kommunenummer_IkkeFunnet,
    Enhetsnummer_IkkeFunnet,
    Enhetsnummer_Funnet;

    override fun fieldName(): String {
        return "routingStep"
    }

    override fun value(): Any {
        return this.toString()
    }
}